src/app/Services/openmrs-api/save-form.service.ts

```diff
@@ -1,170 +1,188 @@
-        this.newValueReference.next(valueReference);
+    const body = {
+      name: name,
+      version: version,
+      published: published || false,
+      description: description || ''
+    };
+    if (encounterType) {
+      body['encounterType'] = encounterType;
     }
+    return this.http
+      .post(`${this.baseUrl}/ws/rest/v1/form`, body, { headers: this.headers })
+      .pipe(map((res) => JSON.parse(JSON.stringify(res))));
+  }
 
-    getNewValueReference() {
-        return this.newValueReference.asObservable();
-       }
-
-    setNewResourceUUID(newResourceUUID) {
-        this.newResourceUUID.next(newResourceUUID);
-    }
+  //////////// get and set new resource uuid and valueReference ////////
+  setNewValueReference(valueReference) {
+    this.newValueReference.next(valueReference);
+  }
 
-    getNewResourceUUID() {
-        return this.newResourceUUID.asObservable();
-    }
+  getNewValueReference() {
+    return this.newValueReference.asObservable();
+  }
 
-    getNewEncounterType() {
-        return this.newEncounterType.asObservable();
-    }
-    setNewFormUUID(newUUID) {
-        this.newUUID.next(newUUID);
-    }
-    getNewFormUUID() {
-        return this.newUUID.asObservable();
-    }
+  setNewResourceUUID(newResourceUUID) {
+    this.newResourceUUID.next(newResourceUUID);
+  }
 
-    setNewFormName(name: string) {
-        return this.formName.next(name);
-    }
+  getNewResourceUUID() {
+    return this.newResourceUUID.asObservable();
+  }
 
-    getNewFormName() {
-        return this.formName.asObservable();
-    }
+  getNewEncounterType() {
+    return this.newEncounterType.asObservable();
+  }
+  setNewFormUUID(newUUID) {
+    this.newUUID.next(newUUID);
+  }
+  getNewFormUUID() {
+    return this.newUUID.asObservable();
+  }
 
-    setNewVersion(version: string) {
-        return this.newVersion.next(version);
-    }
+  setNewFormName(name: string) {
+    return this.formName.next(name);
+  }
 
-    getNewVersion() {
-        return this.newVersion.asObservable();
-    }
+  getNewFormName() {
+    return this.formName.asObservable();
+  }
 
+  setNewVersion(version: string) {
+    return this.newVersion.next(version);
+  }
 
-    setNewDescription(description: string) {
-        return this.newDescription.next(description);
-    }
+  getNewVersion() {
+    return this.newVersion.asObservable();
+  }
 
-    getNewDescription() {
-        return this.newDescription.asObservable();
-    }
+  setNewDescription(description: string) {
+    return this.newDescription.next(description);
+  }
 
-    ///////////////////////////////////////////////////////////////////////
+  getNewDescription() {
+    return this.newDescription.asObservable();
+  }
 
-    publish(uuid) {
-        const body = { published : true };
-        return this.http.post(`${this.baseUrl}/ws/rest/v1/form/${uuid}`, body, {headers: this.headers});
-    }
+  ///////////////////////////////////////////////////////////////////////
```

Comment:
You should remove this. It's not neccessary

Reply:
Done